internal/dot/graph.go

```diff
@@ -252,6 +318,87 @@ func (dg *Graph) addToGroup(r *Result, id CtorID) {
 	group.Results = append(group.Results, r)
 }
 
+// Prune removes elements from the graph that do not have failed results.
+// Removing elements that do not have failing results makes the graph easier to debug, since non-failing
+// nodes and edges can clutter the graph and don't help the user debug.
+func (dg *Graph) PruneSuccess() {
+	dg.pruneCtors()
+	dg.pruneGroups()
```

Comment:
as per offline discussion: let's pass in `dg.Failed.ctors` and `dg.Failed.groups` as parameters here rather than reaching into `dg` constantly.